### ISSUE TITLE
Invalid unnamed union member of struct

### ISSUE THREAD

**Issue by jpnurmi** [dart-lang/ffigen#570]
With the following ffigen config:
```yaml
name: Foo
description: Foo
output: foo.dart
headers:
  entry-points:
    - "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/glob.h"
structs:
  include:
    - "glob_t"
exclude-all-by-default: true
```

And the following `glob_t` declaration in `glob.h` in macOS:
```c
typedef struct {
	size_t gl_pathc;	/* Count of total paths so far. */
	int gl_matchc;		/* Count of paths matching pattern. */
	size_t gl_offs;		/* Reserved at beginning of gl_pathv. */
	int gl_flags;		/* Copy of flags parameter to glob. */
	char **gl_pathv;	/* List of paths matching pattern. */
				/* Copy of errfunc parameter to glob. */
#ifdef __BLOCKS__
	union {
#endif /* __BLOCKS__ */
		int (*gl_errfunc)(const char *, int);
#ifdef __BLOCKS__
		int (^gl_errblk)(const char *, int);
	};
#endif /* __BLOCKS__ */

	/*
	 * Alternate filesystem access methods for glob; replacement
	 * versions of closedir(3), readdir(3), opendir(3), stat(2)
	 * and lstat(2).
	 */
	void (*gl_closedir)(void *);
#if !defined(_POSIX_C_SOURCE) || defined(_DARWIN_C_SOURCE)
	struct dirent *(*gl_readdir)(void *);
#else /* (_POSIX_C_SOURCE && !_DARWIN_C_SOURCE) */
	void *(*gl_readdir)(void *);
#endif /* (!_POSIX_C_SOURCE || _DARWIN_C_SOURCE) */
	void *(*gl_opendir)(const char *);
#if !defined(_POSIX_C_SOURCE) || defined(_DARWIN_C_SOURCE)
	int (*gl_lstat)(const char *, struct stat *);
	int (*gl_stat)(const char *, struct stat *);
#else /* (_POSIX_C_SOURCE && !_DARWIN_C_SOURCE) */
	int (*gl_lstat)(const char *, void *);
	int (*gl_stat)(const char *, void *);
#endif /* (!_POSIX_C_SOURCE || _DARWIN_C_SOURCE) */
} glob_t;
```

Generated with ffigen 7.2.5 or later:

> [WARNING]: Removed All Union Members from UnnamedUnion1(UnnamedUnion1), struct member has an unsupported type.

```dart
// AUTO GENERATED FILE, DO NOT EDIT.
//
// Generated by `package:ffigen`.
import 'dart:ffi' as ffi;

class glob_t extends ffi.Struct {
  @ffi.Size()
  external int gl_pathc;

  @ffi.Int()
  external int gl_matchc;

  @ffi.Size()
  external int gl_offs;

  @ffi.Int()
  external int gl_flags;

  external ffi.Pointer<ffi.Pointer<ffi.Char>> gl_pathv;

  external UnnamedUnion1 unnamed; // <== invalid_field_type_in_struct

  external ffi
          .Pointer<ffi.NativeFunction<ffi.Void Function(ffi.Pointer<ffi.Void>)>>
      gl_closedir;

  external ffi.Pointer<
      ffi.NativeFunction<
          ffi.Pointer<dirent> Function(ffi.Pointer<ffi.Void>)>> gl_readdir;

  external ffi.Pointer<
      ffi.NativeFunction<
          ffi.Pointer<ffi.Void> Function(ffi.Pointer<ffi.Char>)>> gl_opendir;

  external ffi.Pointer<
      ffi.NativeFunction<
          ffi.Int Function(ffi.Pointer<ffi.Char>, ffi.Pointer<stat>)>> gl_lstat;

  external ffi.Pointer<
      ffi.NativeFunction<
          ffi.Int Function(ffi.Pointer<ffi.Char>, ffi.Pointer<stat>)>> gl_stat;
}

class UnnamedUnion1 extends ffi.Opaque {}

class dirent extends ffi.Opaque {}

class stat extends ffi.Opaque {}
```

```
error • foo.dart:21:12 • Fields in struct classes can't have the type 'UnnamedUnion1'. They can only be declared as 'int', 'double', 'Array', 'Pointer', or subtype of 'Struct' or 'Union'. Try using 'int',
          'double', 'Array', 'Pointer', or subtype of 'Struct' or 'Union'. • invalid_field_type_in_struct
```

For reference, the same generated with ffigen 7.2.4 or earlier:

```dart
// AUTO GENERATED FILE, DO NOT EDIT.
//
// Generated by `package:ffigen`.
import 'dart:ffi' as ffi;

class glob_t extends ffi.Struct {
  @ffi.Size()
  external int gl_pathc;

  @ffi.Int()
  external int gl_matchc;

  @ffi.Size()
  external int gl_offs;

  @ffi.Int()
  external int gl_flags;

  external ffi.Pointer<ffi.Pointer<ffi.Char>> gl_pathv;

  external ffi
          .Pointer<ffi.NativeFunction<ffi.Void Function(ffi.Pointer<ffi.Void>)>>
      gl_closedir;

  external ffi.Pointer<
      ffi.NativeFunction<
          ffi.Pointer<dirent> Function(ffi.Pointer<ffi.Void>)>> gl_readdir;

  external ffi.Pointer<
      ffi.NativeFunction<
          ffi.Pointer<ffi.Void> Function(ffi.Pointer<ffi.Char>)>> gl_opendir;

  external ffi.Pointer<
      ffi.NativeFunction<
          ffi.Int Function(ffi.Pointer<ffi.Char>, ffi.Pointer<stat>)>> gl_lstat;

  external ffi.Pointer<
      ffi.NativeFunction<
          ffi.Int Function(ffi.Pointer<ffi.Char>, ffi.Pointer<stat>)>> gl_stat;
}

class dirent extends ffi.Opaque {}

class stat extends ffi.Opaque {}
```

**Comment by jpnurmi** [comment]
We still have this problem with `7.2.11` and `8.0.0`.

**Comment by dcharkes** [comment]
cc @mannprerak2 could you please take a look?

**Comment by dcharkes** [comment]
We have released v8.0.1, please let us know if this solves your issue.

**Comment by mannprerak2** [comment]
No, and it may not be possible to do so in all cases (depends on what the unsupported members are)

But you can define the struct class yourself and ffigen will simply use that reference.
See `type-map` and `library-imports` (Both documented in Readme configuration section)
